src/co/elastic/NotificationManager.groovy

```diff
@@ -60,32 +57,44 @@ def analyzeFlakey(Map args = [:]) {
     def flakyTestsWithIssues = [:]
     def genuineTestFailures = []
 
-    if (!flakyReportIdx?.trim()) {
-      error 'analyzeFlakey: did not receive flakyReportIdx data'
+    if (args.containsKey('flakyReportIdx') && args.containsKey('jobName')) {
+      error('Please pass either `jobName` or `flakyReportIdx` but not both')
```

Comment:
I'm a bit skeptical with this particular validation, since this is a breaking change. 

We are forced to change the consumers anyway, what do you think if we remove entirely the flakyReportIdx argument and make the `jobName` as mandatory?

Reply:
The idea here was to avoid breaking any pipelines during the transition period. If you think that won't be a problem, we can definitely simplify this.

Reply:
I missed your previous comment, sorry about that.

https://github.com/elastic/apm-pipeline-library/blob/ce21387c1d2157a5528321c3719f45bc4e814485/vars/notifyBuildResult.groovy#L193-L201 will not fail but warn the stage only, so I think we are safe even if we force the `jobName`.

what do you think?

Reply:
I think you make a great point here and I prefer your approach. I'll back out my changes and we'll rely on the handling in `notifyBuildResult` to do the right thing. Thanks for raising this issue!

Reply:
I've backed my changes out. Thanks for the discussion here. It was really helpful!